internal/core/src/index/json_stats/JsonKeyStats.h

```diff
@@ -0,0 +1,649 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+#include <boost/filesystem.hpp>
+
+#include "index/InvertedIndexTantivy.h"
+#include "common/jsmn.h"
+#include "mmap/ChunkedColumnInterface.h"
+#include "arrow/api.h"
+#include "index/json_stats/utils.h"
+#include "index/json_stats/bson_inverted.h"
+#include "index/json_stats/parquet_writer.h"
+#include "index/json_stats/bson_builder.h"
+#include "common/bson_view.h"
+#include "index/SkipIndex.h"
+
+namespace milvus::index {
+class JsonKeyStats : public ScalarIndex<std::string> {
```

Comment:
if we are renaming this feature as json shredding, how about we rename this class/folder to json_shredding as well? it would be confusing in the future when the feature name mismatch with the class name.